AxonDeepSeg/segment.py

```diff
@@ -23,223 +23,279 @@
-    border effects but more time to perform the segmentation.
-    :param config: dict containing the configuration of the network
-    :param resolution_model: the resolution the model was trained on.
-    :param segmented_image_prefix: the prefix to add before the segmented image.
-    :param verbosity_level: Level of verbosity. The higher, the more information is given about the segmentation
-    process.
-    :return: Nothing.
-    '''
-
-    if os.path.exists(path_testing_image):
-
-        # Extracting the image name and its folder path from the total path.
-        tmp_path = path_testing_image.split('/')
-        acquisition_name = tmp_path[-1]
-        path_acquisition = '/'.join(tmp_path[:-1])
-
-        # Performing the segmentation
-        segmented_image_name = segmented_image_prefix + acquisition_name
-        axon_segmentation(path_acquisitions_folders=path_acquisition, acquisitions_filenames=[acquisition_name],
-                          path_model_folder=path_model, config_dict=config, ckpt_name='model',
-                          inference_batch_size=1, overlap_value=overlap_value,
-                          segmentations_filenames=segmented_image_name,
-                          resampled_resolutions=resolution_model, verbosity_level=verbosity_level,
-                          acquired_resolution=acquired_resolution,
-                          prediction_proba_activate=False, write_mode=True)
-
-        if verbosity_level >= 1:
-            print "Image {0} segmented.".format(path_testing_image)
-
-    else:
-
-        print "The path {0} does not exist.".format(path_testing_image)
-
-
-
-    return None
+					overlap_value, config, resolution_model, segmented_image_prefix,
+				  acquired_resolution = 0.0, verbosity_level=0):
+
+	'''
+	Segment the image located at the path_testing_image location.
+	:param path_testing_image: the path of the image to segment.
+	:param path_model: where to access the model
+	:param overlap_value: the number of pixels to be used for overlap when doing prediction. Higher value means less
+	border effects but more time to perform the segmentation.
+	:param config: dict containing the configuration of the network
+	:param resolution_model: the resolution the model was trained on.
+	:param segmented_image_prefix: the prefix to add before the segmented image.
+	:param verbosity_level: Level of verbosity. The higher, the more information is given about the segmentation
+	process.
+	:return: Nothing.
+	'''
+
+	if os.path.exists(path_testing_image):
+
+		# Extracting the image name and its folder path from the total path.
+		tmp_path = path_testing_image.split('/')
+		acquisition_name = tmp_path[-1]
+		path_acquisition = '/'.join(tmp_path[:-1])
+
+		# Get type of model we are using
+		tmp_path, selected_model = os.path.split(path_model)
+
+		# If the input image is not png
+		if ((acquisition_name.endswith(".jpeg")) or (acquisition_name.endswith(".jpg")) or (acquisition_name.endswith(".tif"))):  
```

Comment:
if not jpeg OR jpg OR tiff, the program will crash. You need to handle all cases